### ISSUE TITLE
Calling BLAS and negative step views

### ISSUE THREAD

**Issue by ghost** [JuliaLang/julia#39836]
```using LinearAlgebra, Test
M = [2. 3.; 0. 0.] 
a = [1.,2.,3.]

v = view(a,3:-1:2)
@test M * v ≈ M * Vector(v)
```
should pass but fails
```Evaluated: [9.0, 0.0] ≈ [12.0, 0.0]```

Found while looking at #37767, present on a recent 1.7.0-DEV, and 1.4.1.

Looking at LinearAlgebra/blas.jl, it seems there might be more functions which do not correctly handle negative stride yet, that is, the case of a StridedVector being a SubArray. 

For the above example, there should be a fix similar to PR #39751. If so, one should probably add an inline function computing the pointer to the first element (in the sense of having the lowest address in memory) of a StridedVector, and use this function  when BLAS.dgemv is called for StridedVectors (and make the change from #39751 use this function, too/take the code from there).

Warning: This is based on the assumption that the step size can be negative in the BLAS function.
Reading [dgemv.f](https://github.com/xianyi/OpenBLAS/blob/f6e4cf2f9dac6f90b36f49659d883767d63791f5/lapack-netlib/BLAS/SRC/dgemv.f), I got the impression that negative strides are supported for dgemv in OpenBLAS StridedVector's. 
Also page 40 of [blas-report.pdf](http://netlib.org/blas/blast-forum/blas-report.pdf) sounds like other BLAS functions should support negative stride for vectors.

**Comment by mbauman** [comment]
What is the BLAS story on negative strides? Is there a consistent one? A quick search hit [this SO question](https://stackoverflow.com/questions/44875236/why-are-non-positive-strides-disallowed-in-the-blas-gemm-family-of-functions) which I had apparently already seen and upvoted at some point in the past.

**Comment by dagbr** [comment]
Good point! I have edited the description to focus on this specific case. (However, my impression from a quick look at the report linked above is that _vectors_ with negative stride might be fine for some other BLAS functions, too. I don't know how closely various implementations comply to the report.)

**Comment by vitorn** [comment]
Another example of this problem with negative stride (using Julia 1.6.3):
```
using LinearAlgebra
h = [1.0, 2.0]
x = [1.0, 2.0, 3.0]
xv = view(x, 2:-1:1)
dot(h, xv) 
7.0
```
but
`
dot(h, x[2:-1:1])
4.0
```
If at least one of the vectors has only integer elements, the result is correct.  

The problem also happens if you use `h'*xv`, without loading `LinearAlgebra`.

**Comment by N5N3** [comment]
Should be fixed after #42957.
